### PR TITLE
Sundials

### PR DESCRIPTION
**Pull request by bscollin**
Unit tests pass with and without sundials enabled.

In order for this to work with sundials:
* Download cvode:  https://computation.llnl.gov/projects/sundials/download/cvode-2.9.0.tar.gz
* Unzip tarball
* Create build directory
* Change into build directory and run cmake:  cmake -DFCMIX_ENABLE=ON -DCMAKE_INSTALL_PREFIX={install path}  {source path}
* make -jN install
* Then when building MPACT or Futility (or VERA or Mongoose) you need to add -DTPL_ENABLE_SUNDIALS=ON -DSUNDIALS_LIBRARY_DIRS={install path}

### CODE REVIEW

**Comment by bscollin** [comment]
I've done Futility, MPACT, and Mongoose builds with and without Sundials and Futility unit tests pass.  Also had a mixture of debug and release buidls.

File: src/ODESolverTypes.f90 [hunk 1]
```diff
@@ -189,6 +220,146 @@ SUBROUTINE odesolver_sub_absintfc(solver)
...
+        IF(n < 1) THEN
+          CALL eODESolverType%raiseError('Incorrect input to '// &
+            modName//'::'//myName//' - Number of values (n) must be '// &
+              'greater than 0!')
+        ELSE
+          solver%n=n
+        ENDIF
+
+        IF(solvetype == THETA_METHOD) THEN
+          !Error... unsupported option
```
Comment (stimpsonsg) [inline src/ODESolverTypes.f90]: Do we want to fill in the error here?

File: src/ODESolverTypes.f90 [hunk 2]
```diff
@@ -189,6 +220,146 @@ SUBROUTINE odesolver_sub_absintfc(solver)
 !> @param solver The ode solver to act on
 !> @param Params A parameter list with input options
 !>
```
Comment (stimpsonsg) [inline src/ODESolverTypes.f90]: If we want to have all the doxygen data up to date, should add f.

File: src/ODESolverTypes.f90 [hunk 3]
```diff
@@ -189,6 +220,146 @@ SUBROUTINE odesolver_sub_absintfc(solver)
...
+      CALL FCVODE(REAL(TF,C_DOUBLE),REAL(ttmp,C_DOUBLE),solver%ytmp,INT(1,C_INT), ierr)
+      CALL yf%set(solver%ytmp)
+#endif
+    ENDSUBROUTINE step_ODESolverType_Sundials
+!
+!-------------------------------------------------------------------------------
+!> @brief Initializes the ODE Solver Type with a parameter list
+!>
+!> @param solver The ode solver to act on
+!> @param Params A parameter list with input options
```
Comment (stimpsonsg) [inline src/ODESolverTypes.f90]: Same about f here

File: src/ODESolverTypes.f90 (hunk 1 above)
Comment (bscollin) [inline src/ODESolverTypes.f90]: addressed

File: src/ODESolverTypes.f90 (hunk 2 above)
Comment (bscollin) [inline src/ODESolverTypes.f90]: addressed

File: src/ODESolverTypes.f90 (hunk 3 above)
Comment (bscollin) [inline src/ODESolverTypes.f90]: addressed

**Comment by bscollin** [comment]
Addressed Shane's comments.  Merging with master.

**Pull request merged**